src/System.Text.RegularExpressions/tests/Performance/Perf.Regex.Cache.cs

```diff
@@ -0,0 +1,67 @@
+﻿using Microsoft.Xunit.Performance;
+using Xunit;
+
+namespace System.Text.RegularExpressions.Tests
+{
+    public class Perf_Regex_Cache
+    {
+        private const int N = 40_000;
```

Comment:
Given the data I posted (where 99% caches are size 15 and nearly always not full) we should make sure we include and measure is a variation of this that uses size 15 and has only a few (some number less than 15) unique regexes. I would aslo suggest another one where there is only 1 unique regex, in order to stress the "MRU" case.

I appreciate your patience on the endless PR but this is very sensitive code to change. It is essential we do not regress the (literally) 99% case when improving perf for a few customers with very large caches.

Reply:
Ok, correcting it. I will be back with tests results. Testing it on various ways to avoid the lock there.